fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualsProcessingServiceImpl.java

```diff
@@ -143,13 +154,34 @@ public void addPeriodicAccruals(@NotNull LocalDate tillDate, @NotNull Loan loan)
     public void addAccruals(@NotNull LocalDate tillDate) throws JobExecutionException {
         List<Loan> loans = loanRepositoryWrapper.findLoansForAddAccrual(AccountingRuleType.ACCRUAL_PERIODIC, tillDate,
                 !isChargeOnDueDate());
+
+        List<Future<?>> loanTasks = new ArrayList<>();
+
+        FineractContext context = ThreadLocalContextUtil.getContext();
+
+        loans.forEach(outerLoan -> {
+            loanTasks.add(taskExecutor.submit(() -> {
+                ThreadLocalContextUtil.init(context);
```

Comment:
Please reset the ThreadLocalContextUtil context once you finished

Reply:
Will do in a followup PR since it doesn't harm anything.